### PR TITLE
Migrate HC_SR04 class member variable initialization to declarations, augment code so that it compiles.

### PR DESCRIPTION
**Pull request by mcsauder**
**Describe problem solved by the proposed pull request**
This PR migrates HC_SR04 class member variable initialization to their respective declarations, formats whitespace, organizes variable ordering, and modifies the code so that it at least compiles, (it is very much still broken, however).

*NOTE* the hc_sr04 driver has been in a disfunctional state since prior to January 2018.  This PR aims to align the driver with other distance sensor drivers, however it could easily be argued that this PR should be modified to deprecate the hc_sr04 driver entirely.

**Describe your preferred solution**
Ultimately all distance sensor drivers should inherit from a common base class, and this PR makes incremental steps toward that objective.

**Describe possible alternatives**
Deprecate this particular driver entirely.

**Additional context**
See Issue #9279.
*NOTE:* This driver is badly broken... A decision should be made about whether or not it should be fixed or deprecated.

https://www.mouser.com/ds/2/813/HCSR04-1022824.pdf

Please let me know if you have any questions on this PR.  Thanks!

-Mark

### CODE REVIEW

**Comment by mcsauder** [comment]
@teapethu , could you advise me on this driver?  It is currently in a broken state.  I've added code in this PR so that it compiles, but it appears to have been broken for at least the past year and a half.

File: src/drivers/distance_sensor/hc_sr04/hc_sr04.cpp [hunk 1]
```diff
@@ -85,138 +98,105 @@ class HC_SR04 : public cdev::CDev
...
 	virtual ~HC_SR04();
 
-	virtual int 		init();
+	virtual int init();
+
+	void interrupt(unsigned time);
 
-	virtual ssize_t		read(device::file_t *filp, char *buffer, size_t buflen);
-	virtual int			ioctl(device::file_t *filp, int cmd, unsigned long arg);
+	virtual int ioctl(device::file_t *filp, int cmd, unsigned long arg) override;
```
Comment (dagar) [inline src/drivers/distance_sensor/hc_sr04/hc_sr04.cpp]: Things like the ioctl, read, and ringbuffers can usually be gutted entirely. These were used historically to read directly from the device (eg `::open("/dev/distance_sensor0")`), but all of that has moved entirely to orb.

File: src/drivers/distance_sensor/hc_sr04/hc_sr04.cpp [hunk 2]
```diff
@@ -782,54 +731,47 @@ info()
...
 
-static int sonar_isr(int irq, void *context)
-{
-	unsigned time = hrt_absolute_time();
-	/* ack the interrupts we just read */
 
-	if (hc_sr04::g_dev != nullptr) {
-		hc_sr04::g_dev->interrupt(time);
-	}
+// static int sonar_isr(int irq, void *context);
```
Comment (julianoes) [inline src/drivers/distance_sensor/hc_sr04/hc_sr04.cpp]: Why is this commented out? Can it be removed?

File: src/drivers/distance_sensor/hc_sr04/hc_sr04.cpp [hunk 3]
```diff
@@ -225,22 +205,101 @@ HC_SR04::~HC_SR04()
...
 	perf_free(_comms_errors);
 }
 
 int
-HC_SR04::init()
+HC_SR04::collect()
 {
-	int ret = PX4_ERROR;
+	int	ret = -EIO;
+#if 0
```
Comment (julianoes) [inline src/drivers/distance_sensor/hc_sr04/hc_sr04.cpp]: Why is this disabled? I would at least add a comment in code to explain why.

**Comment by mcsauder** [comment]
Closing in favor of #13021!

**Pull request closed without merging**